### PR TITLE
stylo: Don't persist styles on elements not in the document.

### PR DESCRIPTION
**Pull request by heycam**
This is the Servo half of https://bugzilla.mozilla.org/show_bug.cgi?id=1324983, which @emilio has already reviewed.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14776)

### CODE REVIEW

File: components/script/layout_wrapper.rs [hunk 1]
```diff
@@ -199,6 +199,10 @@ impl<'ln> TNode for ServoLayoutNode<'ln> {
             self.node.parent_node_ref().map(|node| self.new_with_this_lifetime(&node))
         }
     }
+
+    fn is_in_doc(&self) -> bool {
+        self.node.is_in_doc()
```
Comment (emilio) [inline components/script/layout_wrapper.rs]: You need to use `unsafe_get` here so this compiles.

Comment (heycam) [inline components/script/layout_wrapper.rs]: Thanks, forgot to test a non-geckolib build.

**Comment by bholley** [comment]
Nit: since the style system operates almost exclusively on elements, methods should go on TElement unless we have an active consumer that needs the method on TNode.

**Pull request merged**